### PR TITLE
Added factor level order feature to the Describe > Two/Three Variable > Pivot Table dialog

### LINKED ISSUE #7620: Add a control of factor level order to the Describe > Two/Three Variable > Pivot Table dialogue
This is a pretty amazing resource for data exploration.
One limitation is that factors appear in alphabetical order.  It ignores the special ordering we may have.

Often that isn't too much of a problem, but sometimes it is!
Note that a factor we may wish to have in the factor-level order may not be given in the dialogue initially.  
The vignette recognises the problem and gives an example where the levels of a factor are defined.
In our case we could just have a receiver on the dialogue that gives the factor name.  Then the order would be the R-Instat order.  Ideally this would be a multiple receiver, but a single would be useful - at least for a start.

Perhaps @lilyclements could have a quick look at the R code needed?  Then @Ivanluv could add the control?  The label might be Factor Levels.  Perhaps there could be a tooltip to say something like.  `Use when default (alphabetical) order of factor levels is inappropriate.`

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #7620 
I have also removed the two variable pivot table dialogue that's no longer being used
@africanmathsinitiative/developers  this is ready for review

### CODE REVIEW

**Comment by Ivanluv** [comment]
Could we have this reviewed and hopefully merged .It contains` .vbproj` changes that are preventing me from opening another PR

File: instat/dlgThreeVariablePivotTable.vb [hunk 1]
```diff
@@ -193,35 +240,58 @@ Public Class dlgThreeVariablePivotTable
...
+                CheckForDuplication(lstColumns, ucrReceiverSelectedVariable, iPosition)
+            End If
 
-        If Not ucrReceiverSelectedVariable.IsEmpty Then
-            CheckForDuplication(lstColumns, ucrReceiverSelectedVariable, iPosition)
-        End If
+            If Not ucrReceiverInitialColumnFactor.IsEmpty AndAlso Not lstColumns.Contains(strColumnVariableName) Then
+                lstColumns.Add(ucrReceiverInitialColumnFactor.GetVariableNames(bWithQuotes:=False))
+                clsConcatenateFunction.AddParameter("col" & iPosition, strColumnVariableName, iPosition:=iPosition, bIncludeArgumentName:=False)
+                iPosition = iPosition + 1
```
Comment (lloyddewit) [inline instat/dlgThreeVariablePivotTable.vb:266]:
```suggestion
                iPosition += 1
```

File: instat/dlgThreeVariablePivotTable.vb [hunk 2]
```diff
@@ -193,35 +240,58 @@ Public Class dlgThreeVariablePivotTable
...
+            End If
 
-        If Not ucrReceiverInitialColumnFactor.IsEmpty AndAlso Not lstColumns.Contains(strColumnVariableName) Then
-            lstColumns.Add(ucrReceiverInitialColumnFactor.GetVariableNames(bWithQuotes:=False))
-            clsConcatenateFunction.AddParameter("col" & iPosition, strColumnVariableName, iPosition:=iPosition, bIncludeArgumentName:=False)
-            iPosition = iPosition + 1
+            If Not ucrReceiverAdditionalRowFactor.IsEmpty AndAlso
+                Not lstColumns.Contains(strRowVariableName) AndAlso ucrChkNumericVariable.Checked Then
+                clsConcatenateFunction.AddParameter("col" & iPosition, strRowVariableName, iPosition:=iPosition, bIncludeArgumentName:=False)
+                iPosition = iPosition + 1
```
Comment (lloyddewit) [inline instat/dlgThreeVariablePivotTable.vb:272]:
```suggestion
                iPosition += 1
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]
@Ivanluv Thank you, very clean and readable code

**Comment by lloyddewit** [comment]
> Could we have this reviewed and hopefully merged .It contains` .vbproj` changes that are preventing me from opening another PR

@rdstern if you can test/approve, then we can merge, thanks

**Review by rdstern — APPROVED** [review]
@lilyclements and @Ivanluv this is now excellent!

**Pull request merged**